lib/virtus/support/type_lookup.rb

```diff
@@ -84,7 +82,7 @@ def determine_type_from_primitive(primitive)
       descendants.select(&:primitive).reverse_each do |descendant|
         descendant_primitive = descendant.primitive
         next unless primitive <= descendant_primitive
-        type = descendant if type.nil? or type.primitive > descendant_primitive
+        type = descendant if type.nil? || type.primitive > descendant_primitive
```

Comment:
@dblock no, `or` is the same as `||`.  Ruby programmers just choose to use `or` only for flow control, but at the end of the day it doesn't matter, he was choosing to be more verbose.